redash/handlers/data_sources.py

```diff
@@ -95,6 +93,9 @@ def post(self):
                                                          name=req['name'],
                                                          type=req['type'],
                                                          options=config)
+
+        models.db.session.commit()
```

Comment:
We probably need to add the `session.commit()` call on many more API calls. I noticed that the tests don't catch this, as they probably use the same session ([this assertion](894da612f4b8af90fbc3ad45ad84aa32062e99bd)  was passing even before this fix).

@jeffwidman @washort any ideas about this? Should we write the tests different?